django/db/migrations/serializer.py

```diff
@@ -244,8 +244,9 @@ def _format(self):
 
 class SetSerializer(BaseSequenceSerializer):
     def _format(self):
-        # Don't use the literal "{%s}" as it doesn't support empty set
-        return "set([%s])"
+        # When value is empty, the empty set should be serialized as "set()",
+        # not "{}" because {} is an empty dict.
+        return '{%s}' if self.value else 'set(%s)'
```

Comment:
The other alternative would be to avoid performing analysis of the migration files.

FWIW I personally find set literals to be better looking and easier to parse particularly in cases where the set doesn't contain many element and the `set([])` boilerplate accounts for half the length of the whole statement.